src/curl.c

```diff
@@ -69,9 +74,113 @@ int swupd_curl_init(void)
 		return -1;
 	}
 
+#ifdef FALLBACK_CAPATHS
+	/* parse and initialize CA paths. */
+	{
+		char *str = strdup(FALLBACK_CAPATHS);
+		char *tok;
+		char *ctx;
+		struct stat st;
+		int i;
+		int sz = 1;
+
+		fallback_capaths = (char **)malloc(sz*sizeof(char *));
+		fallback_capaths[0] = NULL;
+		i = 1;
+
+		tok = strtok_r(str, ":", &ctx);
+		do {
+			if (stat(tok, &st)) {
+				continue;
+			}
+			if ((st.st_mode & S_IFMT) != S_IFDIR) {
+				continue;
+			}
+			if (i == sz) {
+				sz <<= 1;
+				fallback_capaths = (char **)realloc(fallback_capaths, sz*sizeof(char *));
+			}
+			fallback_capaths[i] = strdup(tok);
+			i++;
+
+		} while ((tok = strtok_r(NULL, ":", &ctx)));
+		fallback_capaths_no = i;
+		free(str);
+	}
+#else
+	fallback_capaths = (char **)malloc(sizeof(char *));
+	fallback_capaths[0] = NULL;
+	fallback_capaths_no = 1;
+#endif
+
 	return 0;
 }
 
+static void swupd_curl_test_resume(void);
+
+int swupd_curl_check_network(void) {
+	CURLcode curl_ret;
+	int ret = -1;
+	int i;
+	CURL *c;
+	static int has_network = 0;
+
+	if (has_network) return 0;
+
+	if (!curl) {
+		return -1;
+	}
+
+	c = curl_easy_duphandle(curl);
+	if (!c) {
+		return -1;
+	}
+
+	for (i = 0; i < fallback_capaths_no; i++) {
+		curl_easy_reset(c);
+
+		curl_ret = curl_easy_setopt(c, CURLOPT_URL, version_url);
+		if (curl_ret != CURLE_OK) {
+			goto cleanup;
+		}
+
+		curl_ret = curl_easy_setopt(c, CURLOPT_NOBODY, 1L);
+		if (curl_ret != CURLE_OK) {
+			goto cleanup;
+		}
+
+		if (i) { /* first element represents default CApath, means no explicit setting */
+			curl_ret = curl_easy_setopt(c, CURLOPT_CAPATH, fallback_capaths[i]);
+			if (curl_ret != CURLE_OK) {
+				break;
+			}
+		}
+
+		curl_ret = curl_easy_perform(c);
+
+		switch (curl_ret) {
+			case CURLE_OK:
```

Comment:
Bring these `case` statements back to the same indentation level as the `switch` statement.